robot.py

```diff
@@ -31,6 +35,8 @@ def update(self, timeout=5):
             data = self.queue.get(timeout=timeout)
```

Comment:
I think here we should get everything that is in the queue and mark in a separate log-stream how many messages we have read. During replay we would use this info to return the correct number of messages from the main stream.

Another option allowing for a deterministic replay would be to wait for a certain message to be received from the queue and return everything before. If we use this option the separate log-stream would not be needed.

Reply:
Let's scratch this suggestion as per yesterdays discussion.